int8 ADD_OPTIMIZATION_DATA = 1
int8 PERFORM_OPTIMIZATION = 2
int8 optimization_function
---
bool success
